
# HG changeset patch
# User Hans de Goede <hdegoede@redhat.com>
# Date 1265377116 -3600
# Node ID d983c3ca07d17455217105ac563d3426852567f7
# Parent  1fa67e17b77cf1e1b5b0ff8b47f677f1b557b524
libv4l: Remove unused tinyjpeg-struct stream_begin stream_length members

From: Hans de Goede <hdegoede@redhat.com>

libv4l: Remove unused tinyjpeg-struct stream_begin stream_length members

Priority: normal

Signed-off-by: Hans de Goede <hdegoede@redhat.com>

--- a/v4l2-apps/libv4l/libv4lconvert/tinyjpeg-internal.h	Fri Feb 05 14:31:32 2010 +0100
+++ b/v4l2-apps/libv4l/libv4lconvert/tinyjpeg-internal.h	Fri Feb 05 14:38:36 2010 +0100
@@ -89,9 +89,7 @@
   unsigned int flags;
 
   /* Private variables */
-  const unsigned char *stream_begin, *stream_end;
-  unsigned int stream_length;
-
+  const unsigned char *stream_end;
   const unsigned char *stream;	/* Pointer to the current stream */
   unsigned char *stream_filtered;
   int stream_filtered_bufsize;
--- a/v4l2-apps/libv4l/libv4lconvert/tinyjpeg.c	Fri Feb 05 14:31:32 2010 +0100
+++ b/v4l2-apps/libv4l/libv4lconvert/tinyjpeg.c	Fri Feb 05 14:38:36 2010 +0100
@@ -2098,19 +2098,13 @@
  */
 int tinyjpeg_parse_header(struct jdec_private *priv, const unsigned char *buf, unsigned int size)
 {
-  int ret;
-
   /* Identify the file */
   if ((buf[0] != 0xFF) || (buf[1] != SOI))
     error("Not a JPG file ?\n");
 
-  priv->stream_begin = buf+2;
-  priv->stream_length = size-2;
-  priv->stream_end = priv->stream_begin + priv->stream_length;
+  priv->stream_end = buf + size;
 
-  ret = parse_JFIF(priv, priv->stream_begin);
-
-  return ret;
+  return parse_JFIF(priv, buf + 2);
 }
 
 static const decode_MCU_fct decode_mcu_3comp_table[4] = {

